src/Umbraco.Web.UI.Client/src/views/propertyeditors/fileupload/fileupload.controller.js

```diff
@@ -44,15 +44,22 @@
         .controller('Umbraco.PropertyEditors.FileUploadController', fileUploadController)
         .run(function (mediaHelper, umbRequestHelper, assetsService) {
             if (mediaHelper && mediaHelper.registerFileResolver) {
-
-                //NOTE: The 'entity' can be either a normal media entity or an "entity" returned from the entityResource
+                // NOTE: The 'entity' can be either a normal media entity or an "entity" returned from the entityResource
                 // they contain different data structures so if we need to query against it we need to be aware of this.
                 mediaHelper.registerFileResolver("Umbraco.UploadField", function (property, entity, thumbnail) {
                     if (thumbnail) {
                         if (mediaHelper.detectIfImageByExtension(property.value)) {
-                            //get default big thumbnail from image processor
-                            var thumbnailUrl = property.value + "?rnd=" + moment(entity.updateDate).format("YYYYMMDDHHmmss") + "&width=500&animationprocessmode=first";
-                            return thumbnailUrl;
+
+                            // Get default big thumbnail from image processor
+                            return mediaHelper.getProcessedImageUrl(property.value,
+                                {
+                                    animationprocessmode: "first",
+                                    cacheBusterValue: moment(entity.updateDate).format("YYYYMMDDHHmmss"),
```

Comment:
I feel like it would be better for the server to auto-cache bust this than the responsibility to be on the client?

Reply:
I kind of agree.  But it's only really the client here that knows the entity's update date, and hence what the appropriate cache buster value is.  By the time the request gets to the server it's just a path to an image file.  We could use a random value cache buster at the server, but then we'd be doing it far more than we want to (i.e. if the entity hasn't changed, we want to use the cache if it's available.

Reply:
ok no worries then :)